README.md

```diff
@@ -1,115 +1,192 @@
+- this yocto layer contains the minimal configuration to start working with yocto/poky
+- it builds an image (bootloader + sourcetree + initramfs + kernel + partition table +
+rootfs) which can be used to boot a Raspberry Pi 3b+/4
+- the image can be tested on a on Qemu-based virtual machine
 
-## Prerequisites
 
-**Host OS**
+## prerequisites (host OS)
 
-Ubuntu 22.04 and 20.04 are elegibles as Host OS.
+Both Ubuntu 22.04 and 20.04 are elegible as host OS.
 
-## Installation
 
-1. Install the depencencies
+## installation
+
+Get the apt required packages + repo cloning.
+
+
+### get required packages
+
+The following commands are required to get the apt packages.
 
 ```bash
-sudo apt install gawk wget git-core git diffstat unzip texinfo build-essential chrpath socat cpio python3 python3-pip zstd libsdl1.2-dev xterm make xsltproc docbook-utils fop dblatex xmlto libssl-dev pv
+sudo apt update
+
+sudo apt install \
+    gawk wget git-core git diffstat unzip build-essential chrpath socat cpio \
+    python3 python3-pip zstd libsdl1.2-dev xterm make xsltproc fop xmlto libssl-dev pv
 
 pip3 install kas==3.0.2
 ```
 
-2. Clone the repo
+### clone the repo
+
+Clone the ideafix repo, then cd to the cloned repo folder.
 
 ```bash
 git clone https://github.com/ERNI-Academy/starterkit-embedded-erni-yocto.git
+cd starterkit-embedded-erni-yocto.git
 ```
 
-## Getting Started
+## generation
 
-1. Checkout layers.
+This project uses KAS to generate and build the image. KAS is a wrapper for the
+yocto bitbake standard tool, that makes things easier for newcomers to yocto. It
+uses yml files as input, and from there it downloads the required repositories, and
+generates the bitbake recipes & output folders.
+
+The generation consists in checking out the _ideafix_ configuration file, and then
+issuing the corresponding build command, that in ideafix can either be for qemu or rPI.
+
+
+### layers checkout
+
+The first thing to do is to checkout the main kas file, in our case, `ideafix.yml`.
 
 ```bash
 kas checkout conf/ideafix.yml
+```
 
-# Additionally you can specify the build folder
-# This is useful keep isolated the qemu and the real builds
-export KAS_BUILD_DIR=qemu
-kas checkout conf/ideafix.yml
-# or
-export KAS_BUILD_DIR=rpi3
-kas checkout conf/ideafix.yml
+This will download all the necessary yocto stuff.
+
+The checkout command does not need to be executed on different sessions, it has
+persistency on the same repo clone folder. It is only required to be re-executed when
+you know an update of the yocto is needed.
+
+
+### image build
+
+The image can be built both for rPI (ARM cross-compiling) and for qemu (native simulation). The
+rPI build generates a file that can be flashed into a microSD disk that can be inserted into 
+the rPI. It is suggested to start with the qemu for the first build, and then the successive
+experimental changes.
+
+The build folder is the one pointed to by the `KAS_BUILD_DIR` environment variable. Examples:
```

Comment:
```suggestion
The `KAS_BUILD_DIR` environment variable configures the build folder. It is highly recommended to isolate the build between the different targets. Thus, it could be used one of the following examples if it is more convenient:
```